docs/pages/database-access/guides/clickhouse-self-hosted.mdx

```diff
@@ -0,0 +1,250 @@
+  <TabItem label="ClickHouse Native protocol">
+
+Configure and start the Teleport Service:
+
+(!docs/pages/includes/database-access/db-configure-start.mdx dbProtocol="clickhouse" dbName="clickhouse" databaseAddress="clickhouse.example.com:9440" !)
+  </TabItem>
+</Tabs>
+
+</TabItem>
+<TabItem label="Kubernetes Cluster">
+  Teleport provides Helm charts for installing the Teleport Database Service in Kubernetes Clusters.
+
+  (!docs/pages/kubernetes-access/helm/includes/helm-repo-add.mdx!)
+
+<Tabs>
+  <TabItem label="ClickHouse HTTP Protocol">
+
+(!docs/pages/includes/database-access/db-helm-install.mdx dbName="clickhouse-http" dbProtocol="clickhouse-http" databaseAddress="clickhouse.example.com:8443" dbName="clickhouse-http" !)
+
+  </TabItem>
+  <TabItem label="clickHouse Native Protocol">
+
+(!docs/pages/includes/database-access/db-helm-install.mdx dbName="clickhouse" dbProtocol="clickhouse" databaseAddress="clickhouse.example.com:9440" dbName="clickhouse" !)
+
+  </TabItem>
+</Tabs>
+
+</TabItem>
+</Tabs>
+
+(!docs/pages/includes/database-access/multiple-instances-tip.mdx !)
+
+## Step 5/5. Connect
+
+Once the Database Service has joined the cluster, log in to see the available
+databases:
+
+
+<Tabs>
+  <TabItem label="ClickHouse HTTP Protocol">
+
+```code
+$ tsh login --proxy=teleport.example.com --user=testuser
+$ tsh db ls
+# Name                            Description Allowed Users Labels  Connect
+# ------------------------------- ----------- ------------- ------- ------------------------------
+# clickhouse-http (user: alice)             [*]           env=dev tsh db connect clickhouse-http
+```
+
+<ScopedBlock scope={["cloud"]}>
+
+```code
+$ tsh login --proxy=mytenant.teleport.sh --user=testuser
+$ tsh db ls
+# Name                            Description Allowed Users Labels  Connect
+# ------------------------------- ----------- ------------- ------- ------------------------------
+# clickhouse-http (user: alice)             [*]           env=dev tsh db connect clickhouse-http
+```
+
+</ScopedBlock>
+
+Start Teleport Local Proxy:
+
+```code
+$ tsh proxy db --tunnel clickhouse-http
+# Started authenticated tunnel for the Clickhouse (HTTP) database "clickhouse-http" in cluster "ice-berg.dev" on 127.0.0.1:59215.
+# To avoid port randomization, you can choose the listening port using the --port flag.
+#
+# Use the following command to connect to the database or to the address above using other database GUI/CLI clients:
+#   $ curl http://localhost:59215/
+```
+
+To test connection you can run the following command:
+
+```code
+$ echo 'select 1;' | curl http://localhost:59215/  --data-binary @-
+# 1
+```
+
+To log out of the database and remove credentials:
+
+```code
+# Remove credentials for a particular database instance.
+$ tsh db logout clickhouse-http
+# Remove credentials for all database instances.
+$ tsh db logout
+```
+
+  </TabItem>
+  <TabItem label="ClickHouse Native Protocol">
+<ScopedBlock scope={["oss", "enterprise"]}>
```

Comment:
I think we can avoid using the `ScopedBlock` here by using a `Var` component for the `--proxy` flag.